### PR TITLE
api/server/httputils: DecodePlatform: improve test-coverage

### PR DESCRIPTION
**Pull request by thaJeztah**
- Use fixtures for the JSON strings
- Add test-cases for invalid / malformed JSON
- Check error-message produced
- Add test for "happy path"

### CODE REVIEW

File: api/server/httputils/form_test.go [hunk 1]
```diff
@@ -111,22 +109,74 @@ func TestInt64ValueOrDefaultWithError(t *testing.T) {
...
+		expectedErr  string
+	}{
+		{
+			doc:         "empty platform",
+			expectedErr: `failed to parse platform: unexpected end of JSON input`,
+		},
+		{
+			doc:          "not JSON",
+			platformJSON: `linux/ams64`,
+			expectedErr:  `failed to parse platform: invalid character 'l' looking for beginning of value`,
```
Comment (thaJeztah) [inline api/server/httputils/form_test.go:126]: FWIW; planning to make some improvements here after this;

- Friendlier messages for invalid / non-JSON
- Considering `os` to be required, but _possibly_ (TBD) `architecture` to be optional,
- :point_up: but: if a `variant` is specified, then architecture is required
- Considering `os.features` to be rejected; we currently don't use it for platform-matching, so it may be better to reject it until we do


Basically make the (from a CLI perspective) accepted format to be;

```console
<os>[(os.version)][/architecture|/architecture/variant]
```

**Review by vvoland — APPROVED** [review]
